controllers/spacerequest/spacerequest_controller.go

```diff
@@ -399,6 +412,125 @@ func (r *Reconciler) deleteExistingSubSpace(logger logr.Logger, subSpace *toolch
+	var namespaceAccess []toolchainv1alpha1.NamespaceAccess
+	for _, namespace := range subSpace.Status.ProvisionedNamespaces {
+		// check if kubeconfig secret exists,
+		// if it doesn't exist it will be created
+		secretList := &v1.SecretList{}
+		secretLabels := client.MatchingLabels{
+			toolchainv1alpha1.SpaceRequestLabelKey:                     spaceRequest.GetName(),
+			toolchainv1alpha1.SpaceRequestNamespaceLabelKey:            spaceRequest.GetNamespace(),
+			toolchainv1alpha1.SpaceRequestProvisionedNamespaceLabelKey: namespace.Name,
+		}
+		if err := memberClusterWithSpaceRequest.Client.List(context.TODO(), secretList, secretLabels, client.InNamespace(spaceRequest.GetNamespace())); err != nil {
+			return errs.Wrap(err, fmt.Sprintf(`attempt to list Secrets associated with spaceRequest %s in namespace %s failed`, spaceRequest.GetName(), spaceRequest.GetNamespace()))
+		}
+
+		kubeConfigSecret := &v1.Secret{}
+		switch {
+		case len(secretList.Items) == 0:
+			// create the secret for this namespace
+			clientConfig, err := r.generateKubeConfig(memberClusterWithSpaceRequest, namespace.Name)
+			if err != nil {
+				return err
+			}
+			clientConfigFormatted, err := clientcmd.Write(*clientConfig)
+			if err != nil {
+				return err
+			}
+			kubeConfigSecret = &v1.Secret{
+				ObjectMeta: metav1.ObjectMeta{
+					GenerateName: spaceRequest.Name + "-",
+					Namespace:    spaceRequest.Namespace,
+					Labels:       secretLabels,
+				},
+				Type: v1.SecretTypeOpaque,
+				StringData: map[string]string{
+					"kubeconfig": string(clientConfigFormatted),
+				},
+			}
+			if err := controllerutil.SetControllerReference(spaceRequest, kubeConfigSecret, r.Scheme); err != nil {
+				return errs.Wrap(err, "error setting controller reference for secret "+kubeConfigSecret.Name)
+			}
+			if err := memberClusterWithSpaceRequest.Client.Create(context.TODO(), kubeConfigSecret); err != nil {
+				return err
+			}
+			logger.Info("Created Secret", "Name", kubeConfigSecret.Name)
+
+			// add provisioned namespace and name of the secret that provides access to the namespace
+			namespaceAccess = append(namespaceAccess, toolchainv1alpha1.NamespaceAccess{
+				Name:      namespace.Name,
+				SecretRef: kubeConfigSecret.Name,
+			})
+
+		case len(secretList.Items) == 1:
+			// a secret is already present for this namespace
+			namespaceAccess = append(namespaceAccess, toolchainv1alpha1.NamespaceAccess{
+				Name:      namespace.Name,
+				SecretRef: secretList.Items[0].Name,
+			})
+
+		case len(secretList.Items) > 1:
+			// some unexpected issue causing to many secrets
+			return fmt.Errorf("invalid number of secrets found. actual %d, expected %d", len(secretList.Items), 1)
+		}
+
+	}
+
+	// update space request status in case secrets for provisioned namespace were created.
+	if len(namespaceAccess) > 0 {
+		spaceRequest.Status.NamespaceAccess = namespaceAccess
+		return memberClusterWithSpaceRequest.Client.Status().Update(context.TODO(), spaceRequest)
+	}
```

Comment:
btw, we should also check if there were some namespaces removed from the tier/Space:
1. the sub-space is provisioned with `appstudio-env` tier that uses `-env` namespace
2. the NSTemplateTier is updated by changing the namespace suffix to `-coolenv`
3. the secret for the original `env` stays there.

It's not critical, because the namespace would be deleted anyway, so can be done later in a separate PR.

Reply:
I'm not sure this is the same use case, but since I'm replacing the `spaceRequest.Status.NamespaceAccess` field every time there is an update to the provisioned namespace of the subSpace (see https://github.com/codeready-toolchain/host-operator/pull/789/files#diff-966ecdc6ac4a4479380c18f5150e3926c6c561e6e1e77bd69a56e1f7be942a21R488-R489) this case should be covered. But maybe I'm wrong.

The "only" case that is not covered is when all the provisioned namespaces of the subSpace are deleted. In that case the `spaceRequest.Status.NamespaceAccess` is not updated, but not sure this is a real use case.

Reply:
my point was that if there is a change in the set of namespaces that are provisioned for the tier, then we won't delete the secret of the deleted namespace that were there before. 
I added the comment to this section because one of the of the solution would be reading the set of namespaces from `spaceRequest.Status.NamespaceAccess` before it gets updated and comparing it with the set of provisioned namespaces in `Space.Status` part, so we would know which namespace was deleted and which secret got obsolete and should be removed.

The second solution would be listing all secrets using the `SpaceRequestLabelKey` and comparing with the expected set of secrets that should be present for the given SpaceRequest in the namespace.

But as I mentioned before - it's not critical in any way, so let's take care of it later.

Reply:
ok got your point now! and yes, deletion of obsolete secrets is not there now, only creation of new secrets (ofcourse ).
Please let me know if you want me to create a Jira for this.
Thanks for the additional details!